### ISSUE TITLE
dashboard footer overlaps pipeline cards

### ISSUE THREAD

**Issue by pivotal-jamie-klassen** [concourse/concourse#2914]
## Bug Report

When writing the "login › can fly login with browser and reuse same browser without CSRF issues" watsjs test, I found myself having to wait 5s for the bottom bar to dismiss before clicking the pause button, otherwise I would get false failures. I realized this was because the watsjs tests run in such a small viewport that the bottom bar (with its positive z-index) blocks clicks on the pipeline cards.

## Steps to Reproduce

visit the non-HD dashboard page, with at least one pipeline, with a small viewport height (like 300px should do the trick), and try to click the pause button on the pipeline card.

## Expected Results

you click the pause button and the pipeline pauses

## Actual Results

while your mouse is moving, the bottom bar pops up and overlaps the button so you have to put your mouse in exactly the right position, wait for the bottom bar to dismiss and then click the button.

## Additional Context

I think we could have the bottom bar not have "position: fixed" on screens with sufficiently small height, and just display below the pipeline cards. Alternatively we could add some padding/margin to the bottom of the main dashboard body, but it might have to be of variable size to account for the fact that the height of the bottom bar varies with the width of the viewport.

## Version Info

* Concourse version: 4.2.1 (master)
* Deployment type (BOSH/Docker/binary): docker-compose
* Infrastructure/IaaS: local
* Browser (if applicable): chrome
* Did this used to work? I think it did, and I broke it when I was aggressively refactoring the untested dashboard page

**Comment by matthewpereira** [comment]
Just adding some thoughts since @zoetian mentioned reviving this issue:

Ideally, instead of wrapping the pipeline tiles in a viewport-sized scroll container which gets padded at the top and bottom to prevent overlapping/blocked elements under the  `position: fixed` top and bottom divs, it would be better to have a flexbox layout that allows for a flexible scroll container sandwiched between the top and bottom divs. This way we never have to worry about complicated Elm conditions and tests.

TLDR, Instead of this:

![Current Fixed Layout(1)](https://user-images.githubusercontent.com/6372810/65798108-d1777680-e13e-11e9-8ec0-047adb145441.png)

We aim for this:

![Proposed Flex Layout(1)](https://user-images.githubusercontent.com/6372810/65798119-d63c2a80-e13e-11e9-9dfe-2462c69bff9c.png)

Then the browser handles everything the way you'd expect whether you're in the middle of the scrollable content or scrolled right to the bottom:
![flexbox demo](https://user-images.githubusercontent.com/6372810/65798137-e48a4680-e13e-11e9-8544-29a53402c06a.gif)

The scope of work to change it is mainly CSS related, but we'd have some Elm-related cruft to remove since the project uses inline styles.
